components/base/sensorcontrolled/velocities.go

```diff
@@ -3,159 +3,37 @@ package sensorcontrolled
 import (
 	"context"
 	"math"
-	"time"
 
 	"github.com/golang/geo/r3"
-	"go.viam.com/utils"
 
 	"go.viam.com/rdk/control"
 )
 
-// startControlLoop uses the control config to initialize a control
-// loop using the controls package and store in on the sensor controlled base struct
-// the sensor base in the controllable interface that implements State and GetState
-// called by the endpoint logic of the control thread and the controlLoopConfig
-// is included at the end of this file.
-func (sb *sensorBase) startControlLoop() error {
-	loop, err := control.NewLoop(sb.logger, sb.controlLoopConfig, sb)
-	if err != nil {
-		return err
-	}
-	if err := loop.Start(); err != nil {
-		return err
-	}
-	sb.loop = loop
-
-	return nil
-}
-
-func (sb *sensorBase) setupControlLoop(linear, angular control.PIDConfig) error {
-	// set the necessary options for a sensorcontrolled base
-	options := control.Options{
-		SensorFeedback2DVelocityControl: true,
-		LoopFrequency:                   10,
-		ControllableType:                "base_name",
-	}
-
-	// check if either linear or angular need to be tuned
-	if linear.NeedsAutoTuning() || angular.NeedsAutoTuning() {
-		options.NeedsAutoTuning = true
-	}
-
-	// combine linear and angular back into one control.PIDConfig, with linear first
-	pidVals := []control.PIDConfig{linear, angular}
-
-	// fully set up the control config based on the provided options
-	pl, err := control.SetupPIDControlConfig(pidVals, sb.Name().ShortName(), options, sb, sb.logger)
-	if err != nil {
-		return err
-	}
-
-	sb.controlLoopConfig = pl.ControlConf
-	sb.loop = pl.ControlLoop
-	sb.blockNames = pl.BlockNames
-
-	return nil
-}
-
-func (sb *sensorBase) updateControlConfig(
-	ctx context.Context, linearValue, angularValue float64,
-) error {
-	// set linear setpoint config
-	if err := control.UpdateConstantBlock(ctx, sb.blockNames[control.BlockNameConstant][0], linearValue, sb.loop); err != nil {
-		return err
-	}
-
-	// set angular setpoint config
-	if err := control.UpdateConstantBlock(ctx, sb.blockNames[control.BlockNameConstant][1], angularValue, sb.loop); err != nil {
-		return err
-	}
-
-	return nil
-}
-
 func (sb *sensorBase) SetVelocity(
 	ctx context.Context, linear, angular r3.Vector, extra map[string]interface{},
 ) error {
 	sb.opMgr.CancelRunning(ctx)
+	ctx, done := sb.opMgr.New(ctx)
+	defer done()
 
-	// set the spin loop to false, so we do not skip the call to SetState in the control loop
-	// this will also stop any active Spin calls
-	sb.setPolling(false)
-
-	if len(sb.controlLoopConfig.Blocks) != 0 {
-		// start a sensor context for the sensor loop based on the longstanding base
-		// creator context, and add a timeout for the context
-		timeOut := 10 * time.Second
-		var sensorCtx context.Context
-		sensorCtx, sb.sensorLoopDone = context.WithTimeout(context.Background(), timeOut)
```

Comment:
this is the main think I'm not sure about removing. there seems to be no issues with the calls interrupting each other from the RC card, but I need to spam it a little and also test from a client script